src/integration/java/org/springframework/data/couchbase/repository/typealias/TypeAliasEntityRepositoryTests.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * The aim of this test is to ensure that each query type can use
+ * the TypeAlias to retrieve an entity
+ *
+ * @author Maxence Labusquiere
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = IntegrationTestApplicationConfig.class)
+public class TypeAliasEntityRepositoryTests {
+
+  private static final String TYPE_ALIAS_VALUE = "MyTypeAlias";
+  private static final String KEY_PARTY = "TypeAliasedParty1";
+
+  @Autowired
+  private RepositoryOperationsMapping repositoryOperationsMapping;
+  @Autowired
+  private IndexManager indexManager;
+  @Autowired
+  private Bucket client;
+
+  private TypeAliasedPartyRepository typeAliasedPartyRepository;
+
+  @Before
+  public void setup() throws Exception {
+    RepositoryFactorySupport factory = new CouchbaseRepositoryFactory(repositoryOperationsMapping, indexManager);
+    client.bucketManager().upsertDesignDocument(createViewDocDesign());
+    client.bucketManager().upsertDesignDocument(createGeoDocDesign());
+
+    typeAliasedPartyRepository = factory.getRepository(TypeAliasedPartyRepository.class);
+  }
+
+  @After
+  public void cleanUp() {
+    try {
+      typeAliasedPartyRepository.deleteById(KEY_PARTY);
+    } catch (DataRetrievalFailureException e) {
+    }
+  }
+
+  @Test
+  public void testTypeAliasOnN1QLQuery() {
+    typeAliasedPartyRepository
+      .save(new TypeAliasedParty(KEY_PARTY, "partyName", "testTypeAliasOnN1QLQuery", null, 1, null));
+    Iterable<TypeAliasedParty> parties = typeAliasedPartyRepository.n1qlFindParties();
+    assertTrue(hasSize(1, parties));
+  }
+
+  @Test
+  public void testTypeAliasOnViewQuery() {
+    typeAliasedPartyRepository
+      .save(new TypeAliasedParty(KEY_PARTY, "partyName", "testTypeAliasOnViewQuery", null, 1, null));
+    Iterable<TypeAliasedParty> parties = typeAliasedPartyRepository.findAll();
+    assertTrue(hasSize(1, parties));
+  }
+
+  @Test
+  public void testTypeAliasOnDimensionalQuery() {
+    final Point location = new Point(1, -2);
+    typeAliasedPartyRepository
+      .save(new TypeAliasedParty(KEY_PARTY, "partyName", "testTypeAliasOnDimensionalQuery", null, 1, location));
+    Iterable<TypeAliasedParty> parties = typeAliasedPartyRepository.findByLocationNear(location, new Distance(10));
+    assertTrue(hasSize(1, parties));
+  }
+
+  private boolean hasSize(int expectedSize, Iterable<TypeAliasedParty> parties) {
+    int actualSize = 0;
+    for (TypeAliasedParty ignore : parties) {
+      actualSize++;
+    }
+    return expectedSize == actualSize;
+  }
+
+  private DesignDocument createViewDocDesign() {
+    String mapFunction =
+      "function (doc, meta) { if(doc._class == \"" + TYPE_ALIAS_VALUE + "\") { emit(doc.id, null); } }";
```

Comment:
This should be handled in the `IndexManager`'s method `buildAllView` which uses the type information as well.